cypress/e2e/widgets-landing.cy.ts

```diff
@@ -0,0 +1,47 @@
+function resetToDefaultLayout() {
+  cy.get('button')
+    .contains('Reset to default')
+    .click()
+    .get('#warning-modal-check')
+    .click()
+    .get("button[data-ouia-component-id='primary-confirm-button']")
+    .click();
+}
+
+describe('Widget Landing Page', () => {
+  it('closes all the widgets', () => {
+    cy.login();
+    cy.visit('/');
+
+    // Reset layout to the default
+    resetToDefaultLayout();
+
+    // Ensure that widgets are open and displayed (Number of items in grid expected to be numDefaultWidgets)
+    const numDefaultWidgets = 9;
+    cy.get(
+      '.widgetLayout > section > div > .react-grid-layout > .react-grid-item'
+    )
+      .its('length')
+      .should('be.eq', numDefaultWidgets)
+      .wait(1500);
+
+    const cardActionsSelector =
```

Comment:
We should avoid using CSS selectors. We should use either TEXT or add IDs to the source code.

This will bit us again when PF updates happen for example.

Reply:
Fundamentally it's the same issue we've been struggling with in IQE for ages. I would strongly support adding test IDs to the elements we interact with in tests.

Reply:
Opened https://issues.redhat.com/browse/RHCLOUD-32770 for tracking